index.js

```diff
@@ -1,244 +1,370 @@
+	 * @returns {{ALL: number, CONSOLE: number, WEB: number, DESKTOP: number}}
+	 */
+	static get Instance() {
+		return {
+			ALL: 0,
+			DESKTOP: 1,
+			CONSOLE: 2,
+			WEB: 4
+		};
+	}
+
+	/**
+	 * Mapping of SteamID types to their characters
+	 * @static
+	 * @returns {object}
+	 */
+	static get TypeChars() {
+		return {
+			[SteamID.Type.INVALID]: 'I',
+			[SteamID.Type.INDIVIDUAL]: 'U',
+			[SteamID.Type.MULTISEAT]: 'M',
+			[SteamID.Type.GAMESERVER]: 'G',
+			[SteamID.Type.ANON_GAMESERVER]: 'A',
+			[SteamID.Type.PENDING]: 'P',
+			[SteamID.Type.CONTENT_SERVER]: 'C',
+			[SteamID.Type.CLAN]: 'g',
+			[SteamID.Type.CHAT]: 'T',
+			[SteamID.Type.ANON_USER]: 'a'
+		};
+	}
+
+	/**
+	 * Mask to be used to get the AccountID out of a 64-bit SteamID
+	 * @static
+	 * @returns {number}
+	 */
+	static get AccountIDMask() { return 0xFFFFFFFF; }
+
+	/**
+	 * Mask to be used to get the instance out of the upper 32 bits of a 64-bit SteamID
+	 * @static
+	 * @returns {number}
+	 */
+	static get AccountInstanceMask() { return 0x000FFFFF; }
+
+	/**
+	 * Flags in SteamID instance for chat type IDs
+	 * @static
+	 * @returns {{Lobby: number, Clan: number, MMSLobby: number}}
+	 */
+	static get ChatInstanceFlags() {
+		return {
+			Clan: (SteamID.AccountInstanceMask + 1) >> 1,
+			Lobby: (SteamID.AccountInstanceMask + 1) >> 2,
+			MMSLobby: (SteamID.AccountInstanceMask + 1) >> 3
+		};
+	}
+
+	/**
+	 * Create a new SteamID object.
+	 * @param {string|BigInt} [input] - BigInt containing 64-bit SteamID, or string containing 64-bit SteamID/Steam2/Steam3 text formats. If omitted, creates a blank SteamID object.
+	 */
+	constructor(input) {
+		this.universe = SteamID.Universe.INVALID;
+		this.type = SteamID.Type.INVALID;
+		this.instance = SteamID.Instance.ALL;
+		this.accountid = 0;
+
+		if (!input) {
+			// Use the default invalid values
+			return;
 		}
 
-		if (typeChar == 'c') {
-			this.instance |= SteamID.ChatInstanceFlags.Clan;
-			this.type = SteamID.Type.CHAT;
-		} else if (typeChar == 'L') {
-			this.instance |= SteamID.ChatInstanceFlags.Lobby;
-			this.type = SteamID.Type.CHAT;
+		let matches;
+		if (typeof input == 'bigint' || (typeof input == 'string' && input.match(/^\d+$/))) {
+			// 64-bit ID
+			let num = BigInt(input);
+			this.accountid = Number(num & BigInt(SteamID.AccountIDMask));
+			this.instance = Number((num >> 32n) & BigInt(SteamID.AccountInstanceMask));
+			this.type = Number((num >> 52n) & 0xFn);
+			this.universe = Number(num >> 56n);
+		} else if ((matches = input.match(/^STEAM_([0-5]):([0-1]):([0-9]+)$/))) {
+			// Steam2 ID
+			let [_, universe, mod, accountid] = matches;
+
+			this.universe = parseInt(universe, 10) || SteamID.Universe.PUBLIC; // If it's 0, turn it into 1 for public
+			this.type = SteamID.Type.INDIVIDUAL;
+			this.instance = SteamID.Instance.DESKTOP;
+			this.accountid = (parseInt(accountid, 10) * 2) + parseInt(mod, 10);
+		} else if ((matches = input.match(/^\[([a-zA-Z]):([0-5]):([0-9]+)(:[0-9]+)?]$/))) {
```

Comment:
`=` ? I'm so confused by this

Reply:
It's basically this, but all on one line:
```js
matches = input.match(...);
if (matches) {
```